src/Features/Core/Portable/UseSystemHashCode/Analyzer.cs

```diff
@@ -0,0 +1,352 @@
+
+        /// <summary>
+        /// Analyzes the containing <c>GetHashCode</c> method to determine which fields and
+        /// properties were combined to form a hash code for this type.
+        /// </summary>
+        public ImmutableArray<ISymbol> GetHashedMembers(ISymbol owningSymbol, IOperation operation)
+        {
+            Debug.Assert(CanAnalyze());
+
+            // Capture a copy of `this` as we're in a struct and need to access it from local
+            // functions in this method.
+            var _this = this;
+
+            if (!(operation is IBlockOperation blockOperation))
+            {
+                return default;
+            }
+
+            // Owning symbol has to be an override of Object.GetHashCode.
+            if (!(owningSymbol is IMethodSymbol method) ||
+                method.Name != nameof(GetHashCode) ||
+                method.Locations.Length != 1 ||
+                method.DeclaringSyntaxReferences.Length != 1 ||
+                !method.Locations[0].IsInSource ||
+                !OverridesSystemObject(method))
+            {
+                return default;
+            }
+
+            // Unwind through nested blocks. This also handles if we're in an 'unchecked' block in C#
+            while (blockOperation.Operations.Length == 1 &&
+                   blockOperation.Operations[0] is IBlockOperation childBlock)
+            {
+                blockOperation = childBlock;
+            }
+
+            // Needs to be of the form:
+            //
+            //      // accumulator
+            //      var hashCode = <initializer_or_hash>
+            //
+            //      // 1-N member hashes mixed into the accumulator.
+            //      hashCode = (hashCode op constant) op Hash(member)
+            //
+            //      // return of the value.
+            //      return hashCode;
+
+            var statements = blockOperation.Operations;
+            if (statements.Length < 3)
+            {
+                return default;
+            }
+
+            // First statement has to be the declaration of the accumulator.
+            // Last statement has to be the return of it.
+            if (!(statements.First() is IVariableDeclarationGroupOperation varDeclStatement) ||
+                !(statements.Last() is IReturnOperation returnStatement))
+            {
+                return default;
+            }
+
+            var variables = varDeclStatement.GetDeclaredVariables();
+            if (variables.Length != 1 ||
+                varDeclStatement.Declarations.Length != 1)
+            {
+                return default;
+            }
+
+            var declaration = varDeclStatement.Declarations[0];
+            if (declaration.Declarators.Length != 1)
+            {
+                return default;
+            }
+
+            var declarator = declaration.Declarators[0];
+            if (declarator.Initializer == null ||
+                declarator.Initializer.Value == null)
```

Comment:
`if (declarator.Initializer?.Value is null)`?

Reply:
done!